File: 0938-range-sum-of-bst/0938-range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
const rangeSumBST = (root, low, high) => {
    let sum = 0;
    const dfs = (root, low, high) => {
        if (!root) return;
        if (root.val >= low && root.val <= high) sum += root.val;
        dfs(root.left, low, high);
        dfs(root.right, low, high);
    }
    dfs(root, low, high);
    return sum;
};

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        def dfs(root, sum):
            if root == None:
                return 0
            elif root.val < low:
                return dfs(root.right, sum)
            elif root.val > high:
                return dfs(root.left, sum)
            elif low <= root.val and root.val <= high:
                return root.val + dfs(root.left, sum) + dfs(root.right, sum)
        return dfs(root, 0)